### PR TITLE
Social media sharing

### PR DESCRIPTION
**Pull request by csiyang**
- updated backend with meta data
- added authentication for journey create, update and publish
- updated backend tests
- updated front end to use the meta data for social media sharing

### CODE REVIEW

File: apps/api-journeys/src/modules/image/image.spec.ts [hunk 1]
```diff
@@ -114,6 +121,106 @@ describe('ImageModule', () => {
...
+    it('returns src from a journey with primaryImageBlock', async () => {
+      const journey: Journey = {
+        id: 'journeyId',
+        title: 'published',
+        published: true,
+        locale: 'hi-IN',
+        themeName: ThemeName.base,
+        themeMode: ThemeMode.light,
+        description: null,
+        primaryImageBlockId: '1'
```
Comment (tataihono) [inline apps/api-journeys/src/modules/image/image.spec.ts:134]:
```suggestion
        primaryImageBlockId: 'primaryImageBlockId'
```

File: apps/api-journeys/src/modules/image/image.spec.ts [hunk 2]
```diff
@@ -114,6 +121,106 @@ describe('ImageModule', () => {
...
+      }
+
+      const primaryImage: Block = {
+        id: '1',
```
Comment (tataihono) [inline apps/api-journeys/src/modules/image/image.spec.ts:138]:
```suggestion
        id: 'primaryImageBlockId',
```

File: apps/api-journeys/src/modules/image/image.spec.ts [hunk 3]
```diff
@@ -114,6 +121,106 @@ describe('ImageModule', () => {
...
```
Comment (tataihono) [inline apps/api-journeys/src/modules/image/image.spec.ts:189]:
```suggestion
        id: 'primaryImageBlockId',
```

File: apps/api-journeys/src/modules/image/image.spec.ts [hunk 4]
```diff
@@ -114,6 +121,106 @@ describe('ImageModule', () => {
...
+    it('returns null when primaryImageBlock does not have src', async () => {
+      const journey: Journey = {
+        id: 'journeyId',
+        title: 'published',
+        published: true,
+        locale: 'hi-IN',
+        themeName: ThemeName.base,
+        themeMode: ThemeMode.light,
+        description: null,
+        primaryImageBlockId: '1'
```
Comment (tataihono) [inline apps/api-journeys/src/modules/image/image.spec.ts:185]:
```suggestion
        primaryImageBlockId: 'primaryImageBlockId'
```

File: apps/journeys-e2e/cypress.json [hunk 5]
```diff
@@ -8,5 +8,6 @@
   "video": true,
   "videosFolder": "../../dist/cypress/apps/journeys-e2e/videos",
   "screenshotsFolder": "../../dist/cypress/apps/journeys-e2e/screenshots",
-  "chromeWebSecurity": false
+  "chromeWebSecurity": false,
+  "defaultCommandTimeout": 50000
```
Comment (tataihono) [inline apps/journeys-e2e/cypress.json:12]: I think we need to swap to production compiles for this. It compiled much quicker as a result.

File: apps/journeys-e2e/src/integration/app.spec.ts [hunk 6]
```diff
@@ -1,9 +1,7 @@
-import { getGreeting } from '../support/app.po'
-
 describe('journeys', () => {
-  beforeEach(() => cy.visit('/'))
+  beforeEach(() => cy.visit('http://localhost:4100/'))
```
Comment (tataihono) [inline apps/journeys-e2e/src/integration/app.spec.ts:2]:
```suggestion
  beforeEach(() => cy.visit('/'))
```

File: apps/journeys-e2e/src/integration/firstStep.spec.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+describe('journeys', () => {
```
Comment (tataihono) [inline apps/journeys-e2e/src/integration/firstStep.spec.ts:1]: can this file get a better name. Like [journeyId].spec.ts to match the nextjs page thats being tested.

```suggestion
describe('JourneyPage', () => {
```

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-e2e/cypress.json (hunk 5 above)
Comment (csiyang) [reply apps/journeys-e2e/cypress.json:12]: Not too sure about how to swap this...

**Review by tataihono — APPROVED** [review]

**Pull request merged**